src/tools/illink/src/linker/Linker/MethodDefinitionExtensions.cs

```diff
@@ -107,7 +107,7 @@ public static void ClearDebugInformation (this MethodDefinition method)
 		/// Returns null if <paramref name="index"/> is not a valid parameter index for <paramref name="method"/>.
 		/// <see cref="GetParameter(MethodDefinition, ParameterIndex)"/> for a non-nullable version if you know the index is valid.
 		/// </summary>
-		public static ParameterProxy? TryGetParameter (this MethodDefinition method, ParameterIndex index)
+		internal static ParameterProxy? TryGetParameter (this MethodDefinition method, ParameterIndex index)
```

Comment:
Please note that we introduced some API rules https://github.com/dotnet/runtime/blob/main/src/tools/illink/src/linker/BannedSymbols.txt

One of them is to not access `MethodDefinition.Parameters` directly - we had way too many bugs around "implicit this" parameter being treated differently by IL versus Metadata versus Cecil layers. So we introduced an abstraction `ParameterIndex` and the API rules try to enforce everything to go through that layer if possible.

Ideally you would adopt the same API rules (for the same reasons) - in which case I would expect you to need these extension methods.

Reply:
I saw the banned apis, that's a nice touch.  And I went through the process of enabling it for UnityLinker as well.  I found most of the apis easy to fix.  However, I must admit the Parameter one was the least straight forward.  I didn't see the value in changing our usages of `.Parameters`.  I ended up disabling all of the `.Parameters` errors due to a combination of reasons.

1) The change in some cases was non-trivial in some cases
2) As far as I could tell our code was correct and there was no value to switching.
3) I saw the linker code itself was pretty liberal about using `#pragma warning disable RS0030 ` on `.Paramters` usages.

That's the back story on why I didn't need to expose these.  I did look at making them public, but it started to spiral to `ParameterProxy`, and `MethodProxy`.  I wasn't sure if you would want all of those made public as well and since we were not calling these methods I made them internal

If you're not opposed to making them and their dependencies public I'm happy to do that.  Maybe down the road I'll look at using these apis instead of sprinkling `#pragma warning disable RS0030 ` all over our code.

Reply:
I'm happy you copied the banned APIs - nice :-)
If you don't need the API, please keep them internal - we can always make them public when necessary.

Yes - we are somewhat fuzzy around the enforcement of the Parameters - basically anything outside the MarkStep ignores it. I think it's a sign of the abstraction needing some more work. But honestly it was complex enough as it is... so I'm a bit hesitant trying to fix it :-)

Reply:
I will leave these methods as internal

Reply:
Makes sense - in our case it is exaggerated by having the analysis code shared between 3 tools and 3 type systems (Cecil, S.R.Metadata, Roslyn) - so having an abstraction which works the same in all 3 was beneficial.